include/zephyr/zbus/zbus.h

```diff
@@ -0,0 +1,653 @@
+
+k_timeout_t _zbus_timeout_remainder(uint64_t end_ticks);
+/** @endcond */
+
+/**
+ * @def ZBUS_OBS_DECLARE
+ * This macro list the observers to be used in a file. Internally, it declares the observers with
+ * the extern statement. Note it is only necessary when the observers are declared outside the file.
+ */
+#define ZBUS_OBS_DECLARE(...) FOR_EACH(_ZBUS_OBS_EXTERN, (;), __VA_ARGS__)
+
+/**
+ * @def ZBUS_CHAN_DECLARE
+ * This macro list the channels to be used in a file. Internally, it declares the channels with the
+ * extern statement. Note it is only necessary when the channels are declared outside the file.
+ */
+#define ZBUS_CHAN_DECLARE(...) FOR_EACH(_ZBUS_CHAN_EXTERN, (;), __VA_ARGS__)
+
+/**
+ * @def ZBUS_OBSERVERS_EMPTY
+ * This macro indicates the channel has no observers.
+ */
+#define ZBUS_OBSERVERS_EMPTY
+
+/**
+ * @def ZBUS_OBSERVERS
+ * This macro indicates the channel has listed observers. Note the sequence of observer notification
+ * will follow the same as listed.
+ */
+#define ZBUS_OBSERVERS(...) __VA_ARGS__
+
+/**
+ * @brief Zbus channel definition.
+ *
+ * This macro defines a channel.
+ *
+ * @param _name The channel's name.
+ * @param _type The Message type. It must be a struct or union.
+ * @param _validator The validator function.
+ * @param _user_data A pointer to the user data.
+ *
+ * @see struct zbus_channel
+ * @param _observers The observers list. The sequence indicates the priority of the observer. The
+ * first the highest priority.
+ * @param _init_val The message initialization.
+ */
+#define ZBUS_CHAN_DEFINE(_name, _type, _validator, _user_data, _observers, _init_val)              \
+	static _type _CONCAT(_zbus_message_, _name) = _init_val;                                   \
+	static K_MUTEX_DEFINE(_CONCAT(_zbus_mutex_, _name));                                       \
+	ZBUS_RUNTIME_OBSERVERS_LIST_DECL(_CONCAT(_runtime_observers_, _name));                     \
+	FOR_EACH_NONEMPTY_TERM(_ZBUS_OBS_EXTERN, (;), _observers)                                 \
+	const _ZBUS_STRUCT_DECLARE(zbus_channel, _name) = {                                        \
+		ZBUS_CHANNEL_NAME_INIT(_name)	 /* Name */                                        \
+			.user_data = _user_data, /* User data */                                   \
+		ZBUS_RUNTIME_OBSERVERS_LIST_INIT(                                                  \
+			_CONCAT(_runtime_observers_, _name)) /* Runtime observer list */           \
+			.message_size = sizeof(_type),	     /* Message size */                    \
+		.message = &_CONCAT(_zbus_message_, _name),  /* Reference to the message */        \
+		.validator = (_validator),		     /* Validator function */              \
+		.mutex = &_CONCAT(_zbus_mutex_, _name),	     /* Channel's Mutex */                 \
+		.observers = (const struct zbus_observer **)(const struct zbus_observer *[]){      \
+			FOR_EACH_NONEMPTY_TERM(ZBUS_REF, (,), _observers)                        \
+				NULL}} /* Static observer list */
```

Comment:
Why not define above:
```
	static const struct zbus_observer * _CONCAT(_zbus_observers_, _name) = {		   \
		FOR_EACH_NONEMPTY_TERM(ZBUS_REF, (,), _observers)				   \
		NULL										   \
	};											   \
```
and here use just:
```
		.observers = &_CONCAT(_zbus_observers_, _name) /* Static observer list */          \
```

Reply:
Added.